pkg/datamover/repository_server_test.go

```diff
@@ -0,0 +1,216 @@
+	kopiaCacheDir      string
+	kopiaLogDir        string
+	kopiaConfigDir     string
+	tlsDir             string
+	serverHost         string
+	serverUsername     string
+	serverPassword     string
+	repositoryUser     string
+	testUsername       string
+	testUserPassword   string
+	repositoryPassword string
+	repoPathPrefix     string
+	fingerprint        string
+}
+
+var _ = Suite(&RepositoryServerSuite{})
+
+func (rss *RepositoryServerSuite) SetUpSuite(c *C) {
+	// Check if kopia binary exists in PATH
+	if !CommandExists("kopia") {
+		c.Skip("Skipping repository server datamover unit test. Couldn't find kopia binary in the path.")
+	}
+
+	// Setting Up Repository Server Address
+	rss.address = fmt.Sprintf("%s:%s", "https://0.0.0.0", strconv.Itoa(rand.IntnRange(50000, 60000)))
+
+	// Setting Up Repository Server User Access
+	rss.serverUsername = "user@localhost"
+	rss.serverPassword = "testPassword"
+	rss.serverHost = "localhost"
+	rss.testUsername = "testuser"
+	rss.testUserPassword = rand.String(8)
+
+	// Setting Up Repository Access
+	rss.repositoryUser = "repositoryUser"
+	rss.repositoryPassword = rand.String(8)
+	rss.repoPathPrefix = path.Join("kopia-int", time.Now().UTC().Format(time.RFC3339), rand.String(5))
+
+	rss.ctx = context.Background()
+
+	// Setting Up Kopia Cache, Log and Config Dir
+	rss.kopiaCacheDir = kopiacmd.DefaultCacheDirectory
+	rss.kopiaLogDir = kopiacmd.DefaultLogDirectory
+	rss.kopiaConfigDir = kopiacmd.DefaultConfigDirectory
+
+	// Setting Up TLS Dir
+	temp := c.MkDir()
+	rss.tlsDir = filepath.Join(temp, "tls-"+rand.String(5))
+}
+
+func (rss *RepositoryServerSuite) setupKopiaRepositoryServer(c *C) {
+	// Setting Up Kopia Repository
+	contentCacheMB, metadataCacheMB := kopiacmd.GetGeneralCacheSizeSettings()
+	c.Log("Creating Kopia Repository...")
+	repoCreateCommandArgs := kopiacmd.RepositoryCommandArgs{
+		CommandArgs: &kopiacmd.CommandArgs{
+			RepoPassword:   rss.repositoryPassword,
+			ConfigFilePath: rss.kopiaConfigDir,
+			LogDirectory:   rss.kopiaLogDir,
+		},
+		CacheDirectory:  rss.kopiaCacheDir,
+		Hostname:        rss.serverHost,
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+		RepoPathPrefix:  rss.repoPathPrefix,
+		Username:        rss.repositoryUser,
+		Location:        testutil.GetDefaultS3CompliantStorageLocation(),
+	}
+	repoCreateCmd, err := kopiacmd.RepositoryCreateCommand(repoCreateCommandArgs)
+	c.Assert(err, IsNil)
+	Command(c, repoCreateCmd...)
+
+	// Setting Up Kopia Repository Server
+	tlsCertFile := rss.tlsDir + ".cert"
+	tlsKeyFile := rss.tlsDir + ".key"
+	serverStartCommandArgs := kopiacmd.ServerStartCommandArgs{
+		CommandArgs: &kopiacmd.CommandArgs{
+			RepoPassword:   "",
+			ConfigFilePath: rss.kopiaConfigDir,
+			LogDirectory:   rss.kopiaLogDir,
+		},
+		ServerAddress:    rss.address,
+		TLSCertFile:      tlsCertFile,
+		TLSKeyFile:       tlsKeyFile,
+		ServerUsername:   rss.serverUsername,
+		ServerPassword:   rss.serverPassword,
+		AutoGenerateCert: true,
```

Comment:
if we are setting `autoGenerateCert` to true, why do need to specify custom generated tls certificates in `TLSCertFile` and `TLSKeyFile` ?

Reply:
We're not creating Certificates Manually anywhere, We're setting values in `TLSCertFile` and `TLSKeyFile` so that the certificates it creates should be `tls.cert` and `tls.key` which we're using later in the code to get fingerprint value. If we won't pass these values it creates files like `tls-XXXXX.crt` and `tls-XXXXX.key`